src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -1842,10 +1868,75 @@ namespace rascal {
                   l_block_idx += l_block_size;
                 }  // for (angular_l)
               }    // for cartesian_idx
-            }
-          }  // if (IsHalfNL)
-        }    // if (this->compute_gradients)
-      }      // for (neigh : center)
+            }      // if (is_center_atom)
+          }        // if (IsHalfNL)
+        }          // if (this->compute_gradients)
+      }            // for (neigh : center)
+
+      // In the case of having several periodic images of other centers in
+      // the environement of center i, we need to sum up
+      // of all these contributions, e.g. ij, ij', ij'' ... where
+      // j primes are periodic images of j. And assign this sum back to all
+      // of these terms.
+      if (this->compute_gradients) {
+        // convinience object to associate every center j within the
+        // environement of center i to its corresponding pair ij and its
+        // periodic images.
+        std::map<int, std::vector<ClusterRefKey<2, ClusterLayer>>>
+            periodic_images_of_center{};
+
+        // find all center atoms within the environment of i
+        for (auto pair : center.pairs()) {
+          auto atom_j = pair.get_atom_j();
+          int atom_tag_j = atom_j.get_atom_tag();
+          if (not manager->is_ghost_atom(pair)) {
+            periodic_images_of_center[atom_tag_j].emplace_back(
```

Comment:
Can the finding of periodic images of the center be encapsulated into its own function? That way it can be used again for other descriptors, since this seems like a fairly general-purpose operation.

Reply:
you mean by own function, a function external to the class ?

Reply:
Yes, so that other representations can use such functionality if they need it.